tdrs-backend/tdpservice/users/api/authorization_check.py

```diff
@@ -65,9 +65,29 @@ def get(self, request, *args, **kwargs):
 
         user_in_valid_group = user.is_ofa_sys_admin or user.is_digit_team
 
-        if (user.hhs_id is not None and user_in_valid_group) or settings.BYPASS_KIBANA_AUTH:
+        if (user.hhs_id is not None and user_in_valid_group) or settings.BYPASS_OFA_AUTH:
             logger.debug(f"User: {user} has correct authentication credentials. Allowing access to Kibana.")
             return HttpResponse(status=200)
         else:
             logger.debug(f"User: {user} has incorrect authentication credentials. Not allowing access to Kibana.")
             return HttpResponse(status=401)
+
+class GrafanaAuthorizationCheck(APIView):
+    """Check if user is authorized to view Grafana."""
+
+    query_string = False
+    pattern_name = "grafana-authorization-check"
+    permission_classes = [IsAuthenticated]
+
+    def get(self, request, *args, **kwargs):
+        """Handle get request and verify user is authorized to access kibana."""
+        user = request.user
+
+        user_in_valid_group = user.is_ofa_sys_admin
+
+        if (user.hhs_id is not None and user_in_valid_group) or settings.BYPASS_OFA_AUTH:
```

Comment:
This feels problematic as it will lock our dev team out of the monitoring tool for prod. Do we want to simply utilize the default administrator user for our use?

Also, perhaps out of scope for this ticket but I know Sentry is going to roll up information from lower environments, will we have a grafana per instance? If not, development team will need a way to view information on staging and lower environments. Perhaps we need to architect this next bit out in the cloud ticket #3046 .

Reply:
The default admin login is still required for this. Perhaps we should only ensure the user is in the correct group and not worry about an HHS ID. That would resolve prod loss of access while also keeping our Grafana instance gated behind our nginx proxy.

I don't see why we couldn't have a single Grafana for all envs. We just need to specify the URL for nginx to proxy pass to it.

Reply:
I guess we can also only have one prometheus instance?